### PR TITLE
PWN-5121 - Fix phone number send error, on create and restore wallet

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5121

## Description of Work

Fix bug with sending requset on create and restore wallet

## Screenshots (Optional)

Screenshot #1 | Screenshot #2 | Screenshot #3
:------------:|:-------------:|:------------:
[Paste Pic 1] | [Paste Pic 2] | [Paste Pic 3]

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/CustomShareRestoreInteractor.kt [hunk 1]
```diff
@@ -27,15 +27,17 @@ class CustomShareRestoreInteractor(
         val temporaryUserPrivateKey = restoreFlowDataLocalRepository.userRestorePrivateKey
             ?: throw RestoreWalletFailure("User restore private key is null")
 
-        if (isResend || userPhoneNumber != restoreFlowDataLocalRepository.userPhoneNumber) {
+        val isNumberEquals = userPhoneNumber == restoreFlowDataLocalRepository.userPhoneNumber
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/CustomShareRestoreInteractor.kt:30]: isNumberEquals -> isNumberAlreadyUsed

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/CustomShareRestoreInteractor.kt [hunk 2]
```diff
@@ -27,15 +27,17 @@ class CustomShareRestoreInteractor(
...
+        val isRestoreWalletRequestSent = restoreFlowDataLocalRepository.isRestoreWalletRequestSent
+        if (isResend || (!isRestoreWalletRequestSent && !isNumberEquals)) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/CustomShareRestoreInteractor.kt:32]: condition (isResend || (!isRestoreWalletRequestSent && !isNumberEquals)) -> to a variable called `isRestoreRequestNeeded`

File: app/src/main/java/org/p2p/wallet/auth/repository/RestoreFlowDataLocalRepository.kt [hunk 3]
```diff
@@ -28,6 +30,7 @@ class RestoreFlowDataLocalRepository {
     var userPhoneNumber: PhoneNumber? = null
         set(value) {
             field = value
+            isRestoreWalletRequestSent = true
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/repository/RestoreFlowDataLocalRepository.kt:33]: you're setting it to true using interactor, don't implicitly set it to true here

File: app/src/main/java/org/p2p/wallet/auth/ui/phone/PhoneNumberEnterPresenter.kt [hunk 4]
```diff
@@ -137,11 +140,14 @@ class PhoneNumberEnterPresenter(
...
         } catch (error: Throwable) {
-            Timber.e(error, "Phone number submission failed")
-            view?.showUiKitSnackBar(messageResId = R.string.error_general_message)
+            if (error is GatewayServiceError) {
+                handleGatewayServiceError(error)
+            } else {
+                Timber.e(error, "Phone number submission failed")
+                view?.showUiKitSnackBar(messageResId = R.string.error_general_message)
+            }
+            restoreWalletInteractor.setIsRestoreWalletRequestSent(false)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/phone/PhoneNumberEnterPresenter.kt:150]: named arg

**Review by gslevinkov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**